### PR TITLE
Refactor workflow to create phylogenetic tree by pathogen

### PR DESCRIPTION
**Pull request by tfrcarvalho**
# Description

Creates a new modal to create phylogenetic trees from a pathogen:
* Modular with elements like:
  * modal (with IDSeq styling decisions), 
  * wizard (for managing and styling a workflow as a sequence of pages) and
  * new generic datatable (eventually to be extended to support all datatables in the site); currently supports search and scrolling on body;
  * new checkbox with updated styling
* The workflow includes:
  * initial page to select from existing trees
  * a page to select samples from the project
  * a page to select other samples
  * custom validation on page continue for table name (for instance)

WIP - What is still missing:
* loading extra info from backend

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. 

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [ ] Login Screen
- [ ] Single Upload Page
- [ ] Batch Upload Page
- [ ] All Projects Page
- [ ] Sample Details Page
- [ ] Report Page

Specific components 

* 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

###Batch Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

###All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

###Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

###Report Page
* Data loads succesfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [ ] I have run through the testing script to make sure current functionality is unchanged
- [ ] I have done relevant tests that prove my fix is effective or that my feature works
- [ ] I have spent time testing out edge cases for my feature
- [ ] I have updated the test script or pull request template if necessary
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/assets/src/components/views/phylo_tree/PhyloTreeByPathCreation.jsx [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    this.phyloTreeHeaders = {
+      name: "Phylogenetic Tree",
+      user: "Creator",
+      last_update: "Last Updated",
+      view: "View"
+    };
+
+    this.projectSamplesHeaders = {
+      name: "Name",
+      host: "Host",
```
Comment (cdebourcy) [inline app/assets/src/components/views/phylo_tree/PhyloTreeByPathCreation.jsx]: For now, maybe comment Host/Tissue/Location out so that it's not shown in the modal. I'll fill it in with the actual data later

File: app/assets/src/components/views/phylo_tree/PhyloTreeChecks.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const MIN_READS = 100;
+
+class PhyloTreeChecks {
+  static passesCreateCondition(ntReads, nrReads) {
+    return ntReads >= MIN_READS && nrReads >= MIN_READS;
```
[the remark this answers is not in the document]
Comment (tfrcarvalho) [reply app/assets/src/components/views/phylo_tree/PhyloTreeChecks.js]: Fixed

**Comment by cdebourcy** [comment]
Thanks for making these changes. A few more things:
(1) 
screen shot 2018-09-18 at 4 40 59 pm
Title on first page in Jenn's mock was "Create Phylogenetic Tree", no? Or if you want to add something about selecting samples, maybe "Select samples to include in phylogenetic tree"?

(2) 
screen shot 2018-09-18 at 4 41 21 pm
Something wrong with the title of the second page...

(3) When I'm in a project and click "View Phylogenies", it shows all the trees, instead of just the trees from that project. (Compare to the behavior in prod.)

**Comment by tfrcarvalho** [comment]
(1) Important to distinguish that this first page is about selecting "samples of the project".  Happy to play around with  the title and/or subtitle as long as that is clear (for instance, Rebecca in the demo, did not fully understood it). I will deploy this branch to staging tomorrow so Jenn can do the last pass.
(2) Fixed
(3) Fixed, but if it is a page per project we should make clear what project it is, and probably allow user to change it.
